include/boost/geometry/algorithms/detail/distance/segment_to_box.hpp

```diff
@@ -58,17 +58,20 @@ namespace detail { namespace distance
 {
 
 
-// TODO: Take strategy
-template <typename Segment, typename Box>
-inline bool intersects_segment_box(Segment const& segment, Box const& box)
+template <typename Segment, typename Box, typename Strategy>
+inline bool intersects_segment_box(Segment const& segment, Box const& box,
+                                   Strategy const& strategy)
 {
-    typedef typename strategy::disjoint::services::default_strategy
+    // TODO: pass strategy
+    auto const s = strategies::relate::services::strategy_converter
         <
-            Segment, Box
-        >::type strategy_type;
+            // This is the only strategy defined in distance segment/box
+            // strategies that carries the information about the spheroid
+            // so use it for now.
```

Comment:
can the umbrella strategy carry the spheroid information?

Reply:
It does hold the spheroid, this is why it is constructed from the side strategy here. An alternative would be to construct it from distance strategy but distance strategies were never passed into relops and setops so this backward compatibility mechanism does not support that. I'm now working on umbrella strategies for distance so this code will be replaced anyway.